packages/climate-ref/src/climate_ref/executor/hpc.py

```diff
@@ -0,0 +1,178 @@
+"""This is HPC-based Executor to use job schedulers.
+
+  If you want to 
+     - run REF under the HPC workflows
+     - run REF in multiple nodes
+
+"""
+# mypy: disable-error-code="import-untyped"
+from typing import Any
+from loguru import logger
+
+from climate_ref.config import Config
+from climate_ref.database import Database
+from climate_ref.models import Execution
+from climate_ref_core.diagnostics import ExecutionDefinition, ExecutionResult
+from climate_ref_core.exceptions import ExecutionError
+from climate_ref_core.executor import execute_locally
+from climate_ref_core.logging import add_log_handler
+from climate_ref_core.slurm import SlurmChecker
+import parsl
+
+from parsl.config import Config as ParslConfig
+from parsl.providers import SlurmProvider
+from parsl.launchers import SrunLauncher
+from parsl.executors import HighThroughputExecutor
+from parsl.addresses import address_by_interface
+from parsl import python_app
+
+
+@python_app
+def _process_run(definition: ExecutionDefinition, log_level: str) -> ExecutionResult:
+    # This is a catch-all for any exceptions that occur in the process
+    try:
+        return execute_locally(definition=definition, log_level=log_level)
+    except Exception:  # pragma: no cover
+        # This isn't expected but if it happens we want to log the error before the process exits
+        logger.exception("Error running diagnostic")
+        # This will kill the process pool
+        raise
+
+class HPCExecutor:
+    """
+    Run diagnostics by submitting a job script
+
+    """
+
+    name = "hpc"
+
+    def __init__( 
+        self,
+        *, 
+        database: Database | None = None,
+        config: Config | None = None,
+        scheduler: str = "slurm", 
```

Comment:
These are all parameters that come via configuration so need to be well documented